### PR TITLE
Consolidating contributing.md into developer docs contributing.rst

### PR DESCRIPTION
**Pull request by nathanrpage97**
## References

#8347 

## Code changes
Basic rough conversion of contributing.md into contributing.rst. Using restructured text directives.

This will still needs a lot of work / input as to how to split up the file. This is a rough version to get that process started.

## User-facing changes
contributing.rst has significantly more information based from markdown file

## Backwards-incompatible changes
None

### CODE REVIEW

**Comment by telamonian** [comment]
Did you use `pandoc` or something to do the markdown => rst conversion, or did you do it by hand?

**Comment by nathanrpage97** [comment]
I just did it by hand since I've been wanting to get more familiar with restructured text for myself.

**Comment by nathanrpage97** [comment]
I put it through pandoc to cleanup some of the linelengths and spacing. When I diff a directly converted pandoc version of the CONTRIBUTING.md and the current one I have, there are no noticable differences.

File: docs/source/developer/contributing.rst [hunk 1]
```diff
@@ -1,19 +1,627 @@
...
+automatically on save.
+
+Submitting a Pull Request Contribution
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Generally, an issue should be opened describing a piece of proposed work
+and the issues it solves before a pull request is opened.
+
+Issue Management
+~~~~~~~~~~~~~~~~
```
Comment (telamonian) [inline docs/source/developer/contributing.rst:66]: This section is at the wrong level

File: docs/source/developer/contributing.rst [hunk 2]
```diff
@@ -1,19 +1,627 @@
...
+.. code:: bash
+
+   jlpm run lighthouse:compare normal.json prod.json
+
+This gives us a report of the relative differences between the audits in
+the two reports:
+
+.. admonition:: Resulting Output
+
+   | normal.json -> prod.json
```
Comment (telamonian) [inline docs/source/developer/contributing.rst:401]: this line and the following report seem to have lost some of their previous formatting (though it's a bit hard to tell from github's "rich" preview)

File: docs/source/developer/contributing.rst [hunk 3]
```diff
@@ -1,19 +1,627 @@
...
+     than 60 children/parent element. A large DOM can increase memory
+     usage, cause longer `style
+     calculations <https://developers.google.com/web/fundamentals/performance/rendering/reduce-the-scope-and-complexity-of-style-calculations>`__,
+     and produce costly `layout
+     reflows <https://developers.google.com/speed/articles/reflow>`__.
+     `Learn
+     more <https://developers.google.com/web/tools/lighthouse/audits/dom-size>`__.
+
+Build and run the stand-alone examples
+''''''''''''''''''''''''''''''''''''''
```
Comment (telamonian) [inline docs/source/developer/contributing.rst:509]: This was a preexisting problem, but this section should be at a higher level

File: docs/source/developer/contributing.rst [hunk 4]
```diff
@@ -1,19 +1,627 @@
...
+.. code:: bash
+
+   conda env create -f docs/environment.yml
+
+Alternatively, you can install the documentation dependencies in an
+existing environment using the following command:
+
+.. code:: bash
+
+   conda env update -n <ENVIRONMENT> -f docs/environment.yml
```
Comment (telamonian) [inline docs/source/developer/contributing.rst:604]:
```suggestion
need to install the documentation dependencies with ``pip``:

.. code:: bash

   pip install -r docs/requirements.txt
```

This section refers to a environment.yml file that no longer exists

File: docs/source/developer/contributing.rst [hunk 5]
```diff
@@ -34,66 +642,179 @@ Writing Style
...
+project, including those you made no changes to. Failing to do so may
+cause issues relating to duplication of shared state.
+
+Specifically, when working with Lumino, you'll probably have to link
+your copy of the ``"@lumino/messaging"`` package (in addition to
+whatever packages you actually made changes to). This is due to
+potential duplication of objects contained in the ``MessageLoop``
+namespace provided by the ``messaging`` package.
+
 User Interface Naming Conventions
```
Comment (telamonian) [inline docs/source/developer/contributing.rst:755]: The "User Interface Naming Conventions" section and all of its subsections should be under the "Writing Document" section

Comment (telamonian) [reply docs/source/developer/contributing.rst:755]: probably put it after the "Writing Style" subsection

**Comment by nathanrpage97** [comment]
Ok I've gone ahead and made those adjustments.

**Comment by blink1073** [comment]
Thanks again @nathanrpage97, and congratulations on your first PR to JupyterLab!

**Comment by nathanrpage97** [comment]
Thanks!

**Pull request merged**